setup.py

```diff
@@ -1,44 +1,50 @@
 """Package Setup script for TFX Addons."""
+import datetime
 import itertools
 import os
-import re
 
 from setuptools import find_namespace_packages, setup
 
+PROJECT_NAME = "tfx-addons"
 
-def _get_version():
-  version_file = os.path.join(os.path.dirname(__file__),
-                              'tfx_addons/__init__.py')
-  with open(version_file, 'r') as fp:
-    version_file_text = fp.read()
 
-  version_match = re.search(
-      r"^__version__ = ['\"]([^'\"]*)['\"]",
-      version_file_text,
-      re.M,
-  )
-  if version_match:
-    return version_match.group(1)
-  else:
-    raise RuntimeError("Unable to find version string.")
+def get_last_commit_time() -> str:
+  string_time = os.getenv("NIGHTLY_TIME").replace('"', "")
+  return datetime.strptime(string_time,
+                           "%Y-%m-%dT%H:%M:%SZ").strftime("%Y%m%d%H%M%S")
 
 
+def get_project_name_version():
+  # Version
+  extracted_version = {}
+  base_dir = os.path.dirname(os.path.abspath(__file__))
+  with open(os.path.join(base_dir, "tfx_addons", "version.py")) as fp:
+    exec(fp.read(), extracted_version)  # pylint: disable=exec-used
+
+  return PROJECT_NAME, extracted_version
+
+
+project_name, version = get_project_name_version()
+inclusive_min_tfx_version = version["INCLUSIVE_MIN_TFX_VERSION"]
+exclusive_max_tfx_version = version["EXCLUSIVE_MAX_TFX_VERSION"]
+
 NAME = "tfx-addons"
-# VERSION = .... Change the version in tfx_addons/__init__.py
 
 TESTS_REQUIRE = ["pytest", "pylint", "pre-commit", "isort", "yapf"]
 
 PKG_REQUIRES = {
     # Add dependencies here for your project. Avoid using install_requires.
     "mlmd_client": ["ml-pipelines-sdk>=1.0.0<2", "ml-metadata>=1.0.0<2"],
```

Comment:
add also the same constraints for ml-pipelines-sdk which is tfx but w less standard components